### PR TITLE
feat: adds 'metric_name' field to internal logs for the tag_cardinality_limit component

### PR DESCRIPTION
**Pull request by nomonamo**
This patch addresses issue #15742 by adding the `metric_name` field to internal logs issued by the tag_cardinality_limit component.

### CODE REVIEW

**Comment by nomonamo** [comment]
I'm not very experienced with Rust and the current approach feels "dirty": I'm aware that cloning is a typical anti-pattern when confronted with ownership issues... I thought I'd submit a patch that compiles and iterate from there 
Any suggestions are welcome 🙏

File: src/transforms/tag_cardinality_limit/mod.rs [hunk 1]
```diff
@@ -100,6 +102,7 @@ impl TagCardinalityLimit {
                     for (key, value) in tags_map.iter_sets() {
                         if self.tag_limit_exceeded(key, value) {
                             emit!(TagCardinalityLimitRejectingEvent {
+                                metric_name: metric_name,
```
Comment (spencergilbert) [inline src/transforms/tag_cardinality_limit/mod.rs:105]:
```suggestion
                                metric_name: &metric_name,
```

File: src/transforms/tag_cardinality_limit/mod.rs [hunk 2]
```diff
@@ -116,6 +119,7 @@ impl TagCardinalityLimit {
                             true
                         } else {
                             emit!(TagCardinalityLimitRejectingTag {
+                                metric_name: metric_name,
```
Comment (spencergilbert) [inline src/transforms/tag_cardinality_limit/mod.rs:122]:
```suggestion
                                metric_name: &metric_name,
```

File: src/transforms/tag_cardinality_limit/mod.rs [hunk 3]
```diff
@@ -90,6 +90,8 @@ impl TagCardinalityLimit {
     }
 
     fn transform_one(&mut self, mut event: Event) -> Option<Event> {
+        let event_clone = event.clone();
+        let metric_name = event_clone.as_metric().name();
         let metric = event.as_mut_metric();
```
Comment (spencergilbert) [inline src/transforms/tag_cardinality_limit/mod.rs:95]:
```suggestion
        let metric = event.as_mut_metric();
        let metric_name = metric.name().to_string();
```

**Review by spencergilbert — COMMENTED** [review]
Thanks @nomonamo - what do you think of this? It's a little bit cleaner.

**Comment by nomonamo** [comment]
> Thanks @nomonamo - what do you think of this? It's a little bit cleaner.

Amazing! I was sure there was a nicer way, this is a lot easier on the eyes 😊 Thanks for taking the time to make suggestions, I learned something today!

I'll update my pull request and set it ready for review. 🙇

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**